slick/src/main/scala/slick/jdbc/HsqldbProfile.scala

```diff
@@ -125,8 +128,160 @@ trait HsqldbProfile extends JdbcProfile {
     }
     override val uuidJdbcType = new UUIDJdbcType {
       override def sqlType = java.sql.Types.BINARY
+
       override def sqlTypeName(sym: Option[FieldSymbol]) = "BINARY(16)"
     }
+    override val localTimeType = new LocalTimeJdbcType {
+      override def sqlType = java.sql.Types.TIME
+
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME(3)"
+
+      override def getValue(r: ResultSet, idx: Int): LocalTime = {
+        // Fixes an issue caused because Hsqldb outputs the time in a non-standard format
+        // not adding trailing zeros to the hours. For example: '2:14:41.421' instead of '02:14:41.421'
+        r.getString(idx) match {
+          case null => null
+          case isoTime if isoTime.indexOf(':') == 2 => LocalTime.parse(isoTime)
+          case isoTime if isoTime.indexOf(':') == 1 => LocalTime.parse(s"0$isoTime")
+          case isoTime => throw new RuntimeException(s"$isoTime is not a valid Hsqldb String")
+        }
+      }
+    }
+
+    override val offsetTimeType = new OffsetTimeJdbcType
+    override val offsetDateTimeType = new OffsetDateTimeJdbcType
+    override val instantType = new InstantJdbcType
+
+
+    /**
+      * HSQLDB uses a non-standard string representation of timestamps. It doesn't pad the hour and offset-hour with zeros.
+      * Although the hour can be handled by DateTimeFormatterBuilder, the offset hour can't. See it's method appendOffset()
+      * So we handle this in two steps: first pad (trim) zeros in the offset, and then use a custom DateTimeFormatterBuilder.
+      */
+    trait HsqldbTimeJdbcTypeWithOffset {
+      protected val timeFormatter: DateTimeFormatter = {
+        new DateTimeFormatterBuilder()
+          .append(DateTimeFormatter.ofPattern("H:mm:ss"))
+          .optionalStart()
+          .appendFraction(ChronoField.NANO_OF_SECOND, 0, 9, true)
+          .optionalEnd()
+          .appendOffset("+HH:MM", "Z")
+          .toFormatter()
+      }
+
+      protected val datetimeFormatter = {
+        new DateTimeFormatterBuilder()
+          .append(DateTimeFormatter.ISO_LOCAL_DATE)
+          .appendLiteral(' ')
+          .append(timeFormatter)
+          .toFormatter()
+      }
+
+      /**
+        * Add offset padding e.g. +1:00 becomes +01:00.
+        */
+      def offsetConvertHsqldbToISO(hsqldbString : String) : String = hsqldbString.takeRight(5).toArray match {
```

Comment:
Why are these needed? They look a bit suspicious to me. Couldn't you just use a `DateTimeFormatter` instead? 

Please add a comment on why a `DateTimeFormatter` cannot be used.

Reply:
The comment on lines 157 to 159 explain why these are needed in addition to a `DateTimeFormatter` This whole trait is for handling that the offsets don't have leading zeros.

Reply:
Ok